tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -4041,7 +4041,7 @@ static Value* ConvertScalarOrVector(CGBuilderTy& Builder, CodeGenTypes &Types,
   llvm::Type *SrcTy = Val->getType();
   llvm::Type *DstTy = Types.ConvertType(DstQualTy);
 
-  DXASSERT(Val->getType() == Types.ConvertType(SrcQualTy), "QualType/Value mismatch!");
+  //DXASSERT(Val->getType() == Types.ConvertType(SrcQualTy), "QualType/Value mismatch!");
```

Comment:
I think I can leave this assert in. I think it was getting hit because the arguments were getting slid. I'll fix it.